lib/tapioca/loaders/loader.rb

```diff
@@ -79,15 +97,41 @@ def load_rails_engines
         end
       end
 
-      sig { returns(T::Array[T.untyped]) }
+      sig { returns(T.untyped) }
+      def rails_autoloader
+        return @rails_autoloader if defined?(@rails_autoloader)
+
+        @rails_autoloader = T.let(Rails.autoloaders.once, T.untyped) if Rails.respond_to?(:autoloaders)
+      end
+
+      sig { params(blk: T.proc.void).void }
+      def with_rails_application(&blk)
+        # Store the current Rails.application object so that we can restore it
+        rails_application = Rails.application
+
+        # Create a new Rails::Application object, so that we can load the engines.
+        # Some engines and the `Rails.autoloaders` call might expect `Rails.application`
+        # to be set, so we need to create one here.
+        unless rails_application
+          Rails.application = Class.new(Rails::Application)
+        end
+
+        blk.call
+      ensure
+        unless rails_application
+          Rails.app_class = Rails.application = nil
+        end
+      end
+
+      T::Sig::WithoutRuntime.sig { returns(T::Array[T.class_of(Rails::Engine)]) }
       def rails_engines
         return [] unless Object.const_defined?("Rails::Engine")
 
         safe_require("active_support/core_ext/class/subclasses")
 
         project_path = Bundler.default_gemfile.parent.expand_path
         # We can use `Class#descendants` here, since we know Rails is loaded
-        Object.const_get("Rails::Engine")
```

Comment:
I'm curious, was there a reason these used to be `const_get`?

Reply:
That is a really good question and I'm not sure why, besides maybe it needed to be that way at some point and nobody ever fixed it? I'm assuming that because we already had the `Object.const_defined?("Rails::Engine")` check in place, this change could have been made earlier. @paracycle might know!